### PR TITLE
57 modify pointrequest after deny

### PR DESCRIPTION
**Pull request by emanreyalp**

### CODE REVIEW

File: app/controllers/evaluations_controller.rb [hunk 1]
```diff
@@ -1,45 +1,75 @@
...
 
   def validate_correct_group
-    evaluation = Evaluation.find(params[:evaluation_id] || params[:id])
-    unauthorized_page unless evaluation.group == current_group
+    @evaluation = Evaluation.find(params[:evaluation_id] || params[:id])
+    unauthorized_page unless @evaluation.group == current_group
   end
 
+  def validate_correct_leader
+    redirect_to root_url unless current_user.leader_of? Group.find(params[:group_id])
```
[the remark this answers is not in the document]
Comment (emanreyalp) [reply app/controllers/evaluations_controller.rb]: Actually I want to ascertain that user is leader of the correct (that belongs to the evaluation) group.
So the logic is bad. :-(

File: app/views/evaluations/show.html.erb [hunk 2]
```diff
@@ -6,15 +6,28 @@
...
+      class: 'uk-button uk-button-primary uk-width-1-3 uk-margin uk-display-block uk-align-center' %>
+    <%= link_to 'Pontozás', group_evaluation_table_path(current_group, @evaluation),
+      class: 'uk-button uk-button-primary uk-width-1-3 uk-margin uk-display-block uk-align-center' %>
+  <% end %>
 
-  <%= button_to "Vissza", group_path(current_group), method: :get, class: "uk-button uk-align-center"%>
+  <% if @evaluation.changeable_entry_request_status? %>
+    <%= link_to 'Belépőigény indoklások', group_evaluation_justifications_edit_path(current_group, @evaluation),
+      class: 'uk-button uk-button-primary uk-width-1-3 uk-margin uk-display-block uk-align-center' %>
+    <%= link_to 'Belépőigény véglegesítése', group_evaluation_entryrequests_path(current_group, @evaluation), method: :post,
```
[the remark this answers is not in the document]
Comment (emanreyalp) [reply app/views/evaluations/show.html.erb]: What would be the correct order?

File: config/routes.rb [hunk 3]
```diff
@@ -43,10 +43,14 @@
     resources :evaluations, only: [:show, :edit, :update] do
       resources :principles, only: [:index, :update, :create, :destroy]
       post '/pointdetails/update', to: 'point_details#update'
+      post '/entryrequests', to: 'evaluations#submit_entry_request'
+      delete '/entryrequests', to: 'evaluations#cancel_entry_request', as: :cancel_entry_request
       post '/entryrequests/update', to: 'entry_requests#update'
       get '/justifications/edit', to: 'justifications#edit'
       post '/justifications/update', to: 'justifications#update'
       get '/table', to: 'evaluations#table'
+      post '/points', to: 'evaluations#submit_point_request'
```
[the remark this answers is not in the document]
Comment (emanreyalp) [reply config/routes.rb]: You are right, I will fix it.

File: db/migrate/20180617081041_evaluations_fix.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# TODO : delete this migration at next deploy (after it was run)
```
[the remark this answers is not in the document]
Comment (emanreyalp) [reply db/migrate/20180617081041_evaluations_fix.rb]: I think this migration is not contain a complex logic and that would be better if we dont store inconsistent database. 
We have not made this mistake before, and I (hope) fixed the logic so we can't make this mistake in the future

File: app/controllers/evaluations_controller.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (emanreyalp) [reply app/controllers/evaluations_controller.rb]: Oh, but with the `validate_correct_group` it's ok. So i delete this.

File: app/views/evaluations/show.html.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (falusi94) [reply app/views/evaluations/show.html.erb]: I think the most user friendly is the following order:
- point related pages
- entry related page
- point request approvement/revoke
- entry request approvement/revoke

File: db/migrate/20180617081041_evaluations_fix.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (falusi94) [reply db/migrate/20180617081041_evaluations_fix.rb]: Okay, I can accept that.

**Comment by emanreyalp** [comment]
[Elutasítás után lehessen módosítani az igényeket](https://trello.com/c/J4epca0z/235-elutas%C3%ADt%C3%A1s-ut%C3%A1n-lehessen-m%C3%B3dos%C3%ADtani-az-ig%C3%A9nyeket)

**Pull request merged**